lib/ansible/utils/display.py

```diff
@@ -279,13 +314,15 @@ def display(self, msg, color=None, stderr=False, screen_only=False, log_only=Fal
 
             fileobj.write(msg2)
 
-            try:
-                fileobj.flush()
-            except IOError as e:
-                # Ignore EPIPE in case fileobj has been prematurely closed, eg.
-                # when piping to "head -n1"
-                if e.errno != errno.EPIPE:
-                    raise
+            # With locks, and the fact that we aren't printing from forks
+            # just write, and let the system flush. Everything should come out peachy
+            # try:
+            #     fileobj.flush()
+            # except IOError as e:
+            #     # Ignore EPIPE in case fileobj has been prematurely closed, eg.
+            #     # when piping to "head -n1"
```

Comment:
we might still need to flush, small messages that fit in the buffer won't make it to user until some other messages pushes the limit and/or the program ends

Reply:
Right now my big concern, is that I know we have stdout deadlock problems.  It's not widespread, but can affect people.  We don't have full control over stdout and stderr, and things like multiprocessing flush them on shutdown.

Because I cannot ensure that we won't get into a deadlock flushing stdout with other code that we don't control, I think it's safest to just not do it ourselves.  Our display is very chatty, so I realistically don't believe we're going to see much delay if any.

I debated for a few moments trying to replace stdout/stderr with a "proxy" that had a lock on flush itself, but that sounds like a recipe for disaster.

Reply:
yeah, proxying can end up being a lot worse than the original issue, just made the note to make sure we all knew 'this is possible' .. but as you said, we write a LOT of info all the time and during shutdown it gets flushed anyways

Reply:
fwiw, just as a precaution I added a flush to `TaskQueueManager.cleanup` so that there is no ambiguity about flushing on shutdown